flutter/lib/src/native/c/sentry_native.dart

```diff
@@ -0,0 +1,389 @@
+        }
+      });
+
+  @override
+  FutureOr<void> pauseAppHangTracking() {}
+
+  @override
+  FutureOr<void> resumeAppHangTracking() {}
+
+  @override
+  FutureOr<void> nativeCrash() {
+    Pointer.fromAddress(1).cast<Utf8>().toDartString();
+  }
+
+  @override
+  FutureOr<SentryId> captureReplay(bool isCrash) {
+    _logNotSupported('capturing replay');
+    return SentryId.empty();
+  }
+}
+
+extension on binding.sentry_value_u {
+  void setNativeValue(String key, binding.sentry_value_u? value) {
+    final cKey = key.toNativeUtf8();
+    if (value == null) {
+      SentryNative.native.value_remove_by_key(this, cKey.cast());
+    } else {
+      SentryNative.native.value_set_by_key(this, cKey.cast(), value);
+    }
+    malloc.free(cKey);
+  }
+
+  binding.sentry_value_u get(String key) {
+    final cKey = key.toNativeUtf8();
+    try {
+      return SentryNative.native.value_get_by_key(this, cKey.cast());
+    } finally {
+      malloc.free(cKey);
+    }
+  }
+
+  T? castPrimitive<T>(SentryLogger logger) {
+    if (SentryNative.native.value_is_null(this) == 1) {
+      return null;
+    }
+    final type = SentryNative.native.value_get_type(this);
+    switch (type) {
+      case binding.sentry_value_type_t.SENTRY_VALUE_TYPE_NULL:
+        return null;
+      case binding.sentry_value_type_t.SENTRY_VALUE_TYPE_BOOL:
+        return (SentryNative.native.value_is_true(this) == 1) as T;
+      case binding.sentry_value_type_t.SENTRY_VALUE_TYPE_INT32:
+        return SentryNative.native.value_as_int32(this) as T;
+      case binding.sentry_value_type_t.SENTRY_VALUE_TYPE_DOUBLE:
+        return SentryNative.native.value_as_double(this) as T;
+      case binding.sentry_value_type_t.SENTRY_VALUE_TYPE_STRING:
+        return SentryNative.native
+            .value_as_string(this)
+            .cast<Utf8>()
+            .toDartString() as T;
+      default:
+        logger(SentryLevel.warning,
+            'SentryNative: cannot read native value type: $type');
+        return null;
+    }
+  }
+}
+
+binding.sentry_value_u? dynamicToNativeValue(
+    dynamic value, SentryLogger logger) {
+  if (value is String) {
+    return value.toNativeValue();
+  } else if (value is int) {
+    return value.toNativeValue();
+  } else if (value is double) {
+    return value.toNativeValue();
+  } else if (value is bool) {
+    return value.toNativeValue();
+  } else if (value is Map<String, dynamic>) {
+    return value.toNativeValue(logger);
+  } else if (value is List) {
+    return value.toNativeValue(logger);
+  } else if (value == null) {
+    return SentryNative.native.value_new_null();
+  } else {
+    logger(SentryLevel.warning,
+        'SentryNative: unsupported data for for conversion: ${value.runtimeType} ($value)');
+    return null;
+  }
+}
+
+extension on String {
+  binding.sentry_value_u toNativeValue() {
+    final cValue = toNativeUtf8();
+    final result = SentryNative.native.value_new_string(cValue.cast());
```

Comment:
is there a chance when `cValue.cast()` throws an exception that `cValue` is never freed?

Reply:
I don't believe cast() can throw

Reply:
but I'll check

Reply:
According to my tests this doesn't throw even with invalid agruments. Also, the generated code has the same pattern, e.g. https://github.com/getsentry/sentry-dart/blob/c328ffcf053d131ae2e276be1d448cf4abd22e88/flutter/lib/src/native/cocoa/binding.dart#L23-L29
```dart

  ffi.Pointer<ObjCSel> _registerName1(String name) {
    final cstr = name.toNativeUtf8();
    final sel = _sel_registerName(cstr.cast());
    pkg_ffi.calloc.free(cstr);
    return sel;
  }
```